src/test/extensions/updateHelper.spec.ts

```diff
@@ -31,177 +33,465 @@ const SPEC = {
+      },
+    ],
+  },
+};
 
-    it("should notify the user if billing is no longer required", () => {
-      const newSpec = _.cloneDeep(SPEC);
-      newSpec.billingRequired = false;
+const INSTANCE = {
+  name: "projects/invader-zim/instances/instance-of-official-ext",
+  createTime: "2019-05-19T00:20:10.416947Z",
+  updateTime: "2019-05-19T00:20:10.416947Z",
+  state: "ACTIVE",
+  config: {
+    name:
+      "projects/invader-zim/instances/image-resizer/configurations/95355951-397f-4821-a5c2-9c9788b2cc63",
+    createTime: "2019-05-19T00:20:10.416947Z",
+    sourceId: "fake-official-source",
+    sourceName: "projects/firebasemods/sources/fake-official-source",
+    source: {
+      name: "projects/firebasemods/sources/fake-official-source",
+    },
+  },
+};
 
-      const loggedLines = updateHelper.displayChangesNoInput(SPEC, newSpec);
+const REGISTRY_INSTANCE = {
+  name: "projects/invader-zim/instances/fake-official-instance",
+  createTime: "2019-05-19T00:20:10.416947Z",
+  updateTime: "2019-05-19T00:20:10.416947Z",
+  state: "ACTIVE",
+  config: {
+    name:
+      "projects/invader-zim/instances/image-resizer/configurations/95355951-397f-4821-a5c2-9c9788b2cc63",
+    createTime: "2019-05-19T00:20:10.416947Z",
+    sourceId: "fake-registry-source",
+    sourceName: "projects/firebasemods/sources/fake-registry-source",
+    source: {
+      name: "projects/firebasemods/sources/fake-registry-source",
+    },
+  },
+};
 
-      const expected = ["", "**Billing is no longer required for this extension.**"];
-      expect(loggedLines).to.eql(expected);
+describe("updateHelper", () => {
+  describe("updateFromLocalSource", () => {
+    let promptStub: sinon.SinonStub;
+    let createSourceStub: sinon.SinonStub;
+    let getInstanceStub: sinon.SinonStub;
+    beforeEach(() => {
+      promptStub = sinon.stub(prompt, "promptOnce");
+      createSourceStub = sinon.stub(extensionsHelper, "createSourceFromLocation");
+      getInstanceStub = sinon.stub(extensionsApi, "getInstance").resolves(INSTANCE);
     });
 
-    it("should display nothing if no relevant fields were changed", () => {
-      const newSpec = _.cloneDeep(SPEC);
-      newSpec.license = "drivers";
-
-      const loggedLines = updateHelper.displayChangesNoInput(SPEC, newSpec);
+    afterEach(() => {
+      promptStub.restore();
+      createSourceStub.restore();
+      getInstanceStub.restore();
+    });
 
-      const expected: string[] = [];
-      expect(loggedLines).to.eql(expected);
+    it("should return the correct source name for a valid local source", async () => {
+      promptStub.resolves(true);
+      createSourceStub.resolves(SOURCE);
+      const name = await updateHelper.updateFromLocalSource(
+        "test-project",
+        "test-instance",
+        ".",
+        SPEC,
+        SPEC.name
+      );
+      expect(name).to.equal(SOURCE.name);
     });
-  });
+    it("should throw an error for an invalid source", async () => {
```

Comment:
Please keep a line break between blocks. (applies throughout - makes it easier to parse)